pkg-py/src/querychat/querychat.py

```diff
@@ -233,6 +237,74 @@ def df_to_html(df: IntoFrame, maxrows: int = 5) -> str:
     return table_html + rows_notice
 
 
+def _get_client_from_env() -> Optional[str]:
+    """Get client configuration from environment variable."""
+    env_client = os.getenv("QUERYCHAT_CLIENT", "")
+    if not env_client:
+        return None
+    return env_client
+
+
+def _create_client_from_string(client_str: str) -> chatlas.Chat:
+    """Create a chatlas.Chat client from a provider-model string."""
+    provider, model = (
+        client_str.split("/", 1) if "/" in client_str else (client_str, None)
+    )
+    # We unset chatlas's envvars so we can listen to querychat's envvars instead
+    with temp_env_vars(
+        {
+            "CHATLAS_CHAT_PROVIDER": provider,
+            "CHATLAS_CHAT_MODEL": model,
+            "CHATLAS_CHAT_ARGS": os.environ["QUERYCHAT_CLIENT_ARGS"],
+        },
+    ):
+        return chatlas.ChatAuto(provider="openai")
+
+
+def _resolve_querychat_client(
+    client: Optional[Union[chatlas.Chat, CreateChatCallback, str]] = None,
+) -> chatlas.Chat:
+    """
+    Resolve the client argument into a chatlas.Chat object.
+
+    Parameters
+    ----------
+    client : chatlas.Chat, CreateChatCallback, str, or None
+        The client to resolve. Can be:
+        - A chatlas.Chat object (returned as-is)
+        - A function that returns a chatlas.Chat object
+        - A provider-model string (e.g., "openai/gpt-4.1")
+        - None (fall back to environment variable or default)
+
+    Returns
+    -------
+    chatlas.Chat
+        A resolved chatlas.Chat object
+
+    """
+    if client is None:
+        client = _get_client_from_env()
+
+    if client is None:
+        # Default to OpenAI with using chatlas's default model
+        return chatlas.ChatOpenAI()
+
+    if callable(client) and not isinstance(client, chatlas.Chat):
+        # Backcompat: support the old create_chat_callback style, using an empty
+        # system prompt as a placeholder.
+        client = client(system_prompt="")
```

Comment:
Similar to https://github.com/posit-dev/querychat/pull/60/files#r2273954592, I wouldn't be opposed to a breaking change (assuming we document it properly)